NebulaWorld/Combat/CombatManager.cs

```diff
@@ -269,6 +270,22 @@ public void OnFactoryLoadFinished(PlanetFactory factory)
         {
             factory.veinPool[i].combatStatId = 0;
         }
+
+        // Clear the combatStat pool
+        var astroId = factory.planet.id;
+        var count = 0;
+        var combatStats = GameMain.data.spaceSector.skillSystem.combatStats;
+        var combatStatCursor = combatStats.cursor;
+        var combatStatbuffer = combatStats.buffer;
+        for (var i = 1; i < combatStatCursor; i++)
+        {
+            if (combatStatbuffer[i].id == i && combatStatbuffer[i].astroId == astroId)
```

Comment:
Should `combatStatbuffer[i].id` be stored in a variable to avoid accessing the array multiple times?